src/main/ui/chat/chat-library.gd

```diff
@@ -6,65 +6,42 @@ Dialog is stored as a set of json resources. This class parses those json resour
 		# can't look up chat events without a chat_id; return an empty array
-		push_warning("Interactable %s does not define a 'chat_id' property." % InteractableManager.get_focused())
+		push_warning("Interactable %s does not define a 'chat_id' property." % focused)
 	else:
+		var chat_id: String = focused.get_meta("chat_id")
 		# open the json file for the currently focused interactable
 		var file := File.new()
 		file.open("res://assets/dialog/%s.json" % chat_id, File.READ)
-		var json_items: Array = parse_json(file.get_as_text())
-		for json_item in json_items:
-			# convert the json items into ChatEvents
-			chat_events.append(_parse_chat_event(json_item))
-	return chat_events
-
+		var json_tree := _parse_json_tree(file.get_as_text())
+		chat_tree.from_dict(json_tree)
+	return chat_tree
 
 """
-Converts a json item into a ChatEvent.
+Parses a json dialog tree from the specified json string.
+
+Our dialog parser needs a dictionary wrapped in an array wrapped in a dictionary. We parse simpler json structures by
+wrapping the parsed objects ourselves. This prevents trivial dialog sequences such as 'Tweet!' from requiring
+two extra json wrapper objects.
 """
-func _parse_chat_event(json_item: Dictionary) -> ChatEvent:
-	var chat_event := ChatEvent.new()
-	if json_item.has("who"):
-		# Ordinary dialog spoken by someone; decorate it appropriately
-		chat_event.name = json_item["who"]
-		chat_event.text = json_item["text"]
-		if _accent_defs.has(chat_event.name):
-			chat_event.accent_def = _accent_defs.get(chat_event.name)
-	else:
-		# Dialog with no speaker; decorate it as a thought bubble
-		chat_event.name = ""
-		chat_event.text = "(%s)" % json_item["text"]
-		chat_event.accent_def = _accent_defs.get("Turbo")
-	
-	match json_item.get("mood"):
-		"default": chat_event.mood = ChatEvent.Mood.DEFAULT
-		"smile0": chat_event.mood = ChatEvent.Mood.SMILE0
-		"smile1": chat_event.mood = ChatEvent.Mood.SMILE1
-		"laugh0": chat_event.mood = ChatEvent.Mood.LAUGH0
-		"laugh1": chat_event.mood = ChatEvent.Mood.LAUGH1
-		"think0": chat_event.mood = ChatEvent.Mood.THINK0
-		"think1": chat_event.mood = ChatEvent.Mood.THINK1
-		"cry0": chat_event.mood = ChatEvent.Mood.CRY0
-		"cry1": chat_event.mood = ChatEvent.Mood.CRY1
-		"sweat0": chat_event.mood = ChatEvent.Mood.SWEAT0
-		"sweat1": chat_event.mood = ChatEvent.Mood.SWEAT1
-		"rage0": chat_event.mood = ChatEvent.Mood.RAGE0
-		"rage1": chat_event.mood = ChatEvent.Mood.RAGE1
-	
-	return chat_event
+func _parse_json_tree(json: String) -> Dictionary:
+	var json_tree: Dictionary
+	var parsed = parse_json(json)
+	if typeof(parsed) == TYPE_DICTIONARY and parsed.has(""):
+		json_tree = parsed
+	elif typeof(parsed) == TYPE_DICTIONARY:
+		# We were given a dictionary corresponding to a dialog line. Wrap it into a tree.
+		json_tree = {"" : [parsed]}
+	elif typeof(parsed) == TYPE_ARRAY:
+		# We were given an array corresponding to a dialog branch. Wrap it into a tree.
+		json_tree = {"" : parsed}
```

Comment:
This could return something crazy like a float and a string; maybe expand this tree to report an error in those cases for symmetry with `ChatEvent.from_dict()`:

`push_error("Invalid json tree: " % json_link)`

Reply:
Now pushes an error.